web/modules/custom/collection/src/Form/JoinCollectionForm.php

```diff
@@ -115,8 +149,12 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
       ]));
     }
 
-    // Check if the user is already a member of the collection.
-    if (Og::isMember($collection, $user)) {
+    $membership = $this->membershipManager->getMembership($collection, $user, [
+      OgMembershipInterface::STATE_PENDING,
+      OgMembershipInterface::STATE_ACTIVE,
+    ]);
```

Comment:
Since this code is duplicated inside this class it would be better to move it to a protected method.